lib/vrl/compiler/src/value.rs

```diff
@@ -72,6 +72,31 @@ impl fmt::Display for Value {
     }
 }
 
+impl From<serde_json::Value> for Value {
+    fn from(json_value: serde_json::Value) -> Self {
+        match json_value {
+            serde_json::Value::Bool(b) => Value::Boolean(b),
+            serde_json::Value::Number(n) => {
+                let float_or_byte = || {
+                    n.as_f64()
+                        .map_or_else(|| Value::Bytes(n.to_string().into()), Value::from)
+                };
+                n.as_i64().map_or_else(float_or_byte, Value::Integer)
+            }
```

Comment:
nit, I'd probably write it something like so:

```suggestion
            serde_json::Value::Number(n) if n.is_i64() => n.as_i64().unwrap().into(),
            serde_json::Value::Number(n) if n.is_f64() => n.as_i64().unwrap().into(),
            serde_json::Value::Number(n) => n.to_string().into(),
```